### ISSUE TITLE
esp_wifi_statis_dump - to file like esp_pm_dump_locks (IDFGH-4424)

### ISSUE THREAD

**Issue by patrick277** [espressif/esp-idf#6255]
**Is your feature request related to a problem? Please describe.**
1. yes: https://github.com/espressif/esp-idf/issues/6220 

**Describe the solution you'd like**
1. enable esp_wifi_statis_dump  to be passed a FILE* ptr
2. a clear description of what the stats mean

1. make it the same as esp_pm_dump_locks

**Additional context**

it would be great to get a description as well for what these mean and the units

I (115836) wifi: BUFFER:
I (115836) wifi: t  max alloc total_alloc total_free fail_fc  fail_oom
I (115836) wifi: 1  32  0     36          36         0        0
I (115846) wifi: 2  32  0     34          34         0        0
I (115846) wifi: 3  32  0     37          37         0        0
I (115856) wifi: 4  32  0     0           0          0        0
I (115866) wifi: 5  0   0     4           4          0        0
I (115866) wifi: 6  0   0     0           0          0        0
I (115876) wifi: 7  32  0     1730        1730       0        0
I (115876) wifi: 8  32  0     191         191        0        0
I (115886) wifi: 9  0   0     0           0          0        0
I (115896) wifi:
I (115896) wifi: LMAC TX:
I (115896) wifi: tx_all   lrc src age to
I (115896) wifi: 85       0   0   0   0
I (115906) wifi: 0        0   0   0   0
I (115906) wifi: 22       0   0   0   0
I (115916) wifi: 0        0   0   0   0
I (115916) wifi: 0        0   0   0   0
I (115916) wifi:
I (115916) wifi: LMAC RX:
I (115926) wifi: ampdu    mdpu     mgmt     ctrl     frag
I (115926) wifi: 968      233      424      30       0
I (115936) wifi:
I (115936) wifi: HW RX:
I (115936) wifi: rx_suc   rx_end   full#1   full#2
I (115946) wifi: 2d34     125c9    1        0
I (115946) wifi:
I (115946) wifi: Hardware:
I (115946) wifi: rxbufbk  rxcckerr ofdmerr  rx_agc   rxsf     rx_abort rx_fcs   fifofull
I (115956) wifi: 1        58fb     58fa     0        51       360      1a8f     0
I (115966) wifi: afull    rx_oth   tkiperr  samebm   ack_int  rts_int  cts_int  rifs_int
I (115976) wifi: 0        1d1      0        0        65c      a37      c4d      0
I (115986) wifi: blockerr hoperr   txrts    txcts    track    trcts    trigger  txhung   panic
I (115986) wifi: 0        0        b        0        33       23       b        2        0
I (115996) wifi:
I (115996) wifi: DIAG:
I (116006) wifi: diag0    diag1    diag2    diag3    diag4    diag5    diag6
I (116006) wifi: 1        0        fa00000  210      586d8001 20000871 0
I (116016) wifi: diag7    diag8    diag9    diag10   diag12   diagsel
I (116026) wifi: 10       0        8ac000   40       0        0
I (116026) wifi:
I (116026) wifi: HMAC RX:
I (116036) wifi: ifx all      data     ctl    mgmt   bcn/probe
I (116036) wifi: sta 461      264      0      174    165
I (116046) wifi: ap  0        0        0      0      0
I (116046) wifi:
I (116046) wifi: ifx auth     assoc    deauth action amsdu
I (116056) wifi: sta 1        1        0      7      0
I (116056) wifi: ap  0        0        0      0      0
I (116066) wifi:
I (116066) wifi: HMAC TX:
I (116066) wifi: sta_tx   ap_tx    psq_tx psq_mc psq_uc ps_uc1 reorder oos   amsdu
I (116076) wifi: 36       0        0      0      0      0       0

**Comment by Alvin1Zhang** [comment]
@patrick277 Thanks for raising this feature request.

**Comment by YouDONG-ESP** [comment]
Hi @patrick277 It's going to be a breaking change and causes problem to other users. 
Do you just want to use this function for #6220?

**Comment by patrick277** [comment]
> 
> 
> Hi @patrick277 It's going to be a breaking change and causes problem to other users.
> Do you just want to use this function for #6220?

I would like have the option to send this data back to my server rather than dump to serial port. to allow remote debugging

**Comment by YouDONG-ESP** [comment]
@patrick277 The thing is, we will not add this feature in our release version because it will bother other users. Meanwhile, I can compile a library for you separatly for temporary debugging.

**Comment by YouDONG-ESP** [comment]
Pls give me your commit ID if you agee with it.

**Comment by YouDONG-ESP** [comment]
@patrick277 Do you still need it?

**Comment by negativekelvin** [comment]
@YouDONG-ESP why does it have to be a breaking change just add a function esp_wifi_status_dump_to_file

**Comment by YouDONG-ESP** [comment]
@negativekelvin  `esp_wifi_status_dump_to_file(uint32_t modules, FILE* stream)` will be a duplicated function with `esp_wifi_statis_dump(uint32_t modules)`.  We may update this API in 5.0

**Comment by negativekelvin** [comment]
@YouDONG-ESP not really duplicated, similar to esp_core_dump_to_flash and esp_core_dump_to_uart and can be done as a non breaking change. If you are asking users for dumps for debugging purposes it should be implemented.

**Comment by jack0c** [comment]
@negativekelvin actually esp_wifi_statis_dump is part of ESP_LOGx. It doesn't have ability to dump anything to flash or file if ESP_LOGx doesn't have the ability.

**Comment by negativekelvin** [comment]
@jack0c yes that is the problem, we are asking for a new api function that doesn't use log and that gives the user control of the data in an easily parsable format.

**Comment by jack0c** [comment]
@negativekelvin the main reason is the wifi lib do not have any support of file operations. I will try to use the file operations from esp-idf.

**Comment by igrr** [comment]
@jack0c I think one way could be for the Wi-Fi library to implement some internal API to pass the diagnostic data from the lib into the esp_wifi component. This structure doesn't need to be public. Then in esp_wifi component you can implement the stable API for users (either using FILE* or something else).

This way you can always change diagnostic structure format without breaking the API.

**Comment by avrono** [comment]
It does not appear this was ever implemented. It would even be useful if there was a similar method just to return a string, so users can do what they want with the output.
